### ISSUE TITLE
[Bug] request.getUrl() doesn't include query

### ISSUE THREAD

**Issue by marcelof01** [getinsomnia/insomnia#1049]
Version 5.16.6
Shell 1.8.4
Node 8.2.1
V8 5.9.211.38
Architecture x64
libcurl/7.59.0 OpenSSL/1.1.0h zlib/1.2.11 nghttp2/1.31.1

OS: Void Linux - Kernel 4.16.12_1 #1 SMP PREEMPT x86_64 GNU/Linux

## Details

I'm making a plugin and I include some calculated hash based in the body and the query of the request. 

It works fine so far, but I noted if I put the GET parameters in the "Query" tab they don't show up in request.getUrl(), as they normally do when I manually write the query string in the text field next to the "Send" button. The query arguments do show up in "request.getParameters()" but that object seems to be a map and I would have to build the query string myself. 

I would like to have access to the actual query string being sent since the calculations made are influenced by the order and format of the parameters in the same.

Maybe there's already another way to access the actual query being sent in the hook? I'm using this:

```javascript
module.exports = context => {
  const request = context.request;
  const url = document.createElement('a');
  url.href  = request.getUrl();
  const host = url.hostname;
  const port = url.port;
  const path = url.pathname;
  const query = url.search.startsWith("?") ? decodeURIComponent(url.search.substring(1)) : "";
  ...
```

I think this is in the edge between feature request, bug, and question...

**Comment by ctcampbell** [comment]
Also have this issue. I need to calculate an HMAC auth header and currently have no way of doing this correctly if users are putting query values in the query tab rather than in the URL bar.

**Comment by danbress** [comment]
I wanted to ping this ticket again, as it surfaced for me but in a different way.  Today a colleague(@JessicaGarson) and I were troubleshooting an issue where we could make a call to our endpoint via Insomnia(got a 200), but if we generated curl code, the curl code would not run successfully(got a 401)

We debugged this and found that if we defined our query parameters in the URL field of Insomnia, then both Insomnia and its generated curl work.  But if we defined our query parameters in the 'query' tab of Insomnia then Insomnia worked(200), but the generated curl did not(401).

When we diffed the curl commands we found the oauth_nonce and oauth_signature, and it seemed like it was related to this ticket.

Happy to create a new ticket "Generated curl does not work if using Oauth1.0 and query parameters defined in 'query' tab" if that is more helpful?
